### PR TITLE
DOC: Update to docstring of pd.DataFrame(dtype)

### LINKED ISSUE #14764: DOC: pd.DataFrame(dtype) arg cannot be list, dict, Series. And None will infer wider type than necessary.
#### Code Sample, a copy-pastable example if possible

```python

# This is a DOCbug simply to document what pd.DataFrame(dtype) currently does.
# It is non-obvious to new users, gives non-obvious error messages, and also behaves differently to read_csv(dtype)

# a) Leaving dtype=None in constructor will infer a wider type than necessary
df_cols = {'year':np.int32, 'month':np.int8}
df = pd.DataFrame(columns=df_cols.keys(), dtype=None, index=range(10), data=-1)
>>> df.dtypes
month    int64
year     int64

# b) The doc doesn't explicitly say a list/dict/Series/array-like is not allowed (and if you
# pass in one the error is not very friendly). Also behaves differently to read_csv(dtype)
df = pd.DataFrame(columns=df_cols.keys(), dtype=np.int32, index=range(10), data=-1)
# df.dtypes shows they're all np.int32
# Fix up dtypes after declaration
for col,coltype in df_cols.items():
    df[col] = df[col].astype(coltype) 

```
#### Problem description

The DataFrame() doc doesn't explicitly say a list/dict/Series/array-like is not allowed (and if you pass in one the error is not very friendly). Also behaves differently to read_csv(dtype).
Leaving dtype=None in constructor will infer a wider type than necessary.
So in general you either set dtype=widest_necessary_type, or dtype=None and then manually fix them up after declaration, by casting with `astype()`

#### Output of ``pd.show_versions()``

<details>
python: 2.7.10.final.0
python-bits: 64
machine: x86_64
processor: i386
byteorder: little
pandas: 0.19.1
numpy: 1.11.2
scipy: 0.18.1
</details>

### PR DESCRIPTION
**Pull request by patniharshit**
- [ ] closes #14764 
 - [ ] tests added / passed
 - [ ] passes ``git diff upstream/master | flake8 --diff``
 - [ ] whatsnew entry

All suggestions to improve this are welcome.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -231,6 +231,23 @@ class DataFrame(NDFrame):
     >>> df3 = DataFrame(np.random.randn(10, 5),
     ...                 columns=['a', 'b', 'c', 'd', 'e'])
 
+    Exceptional Behaviour
```
Comment (jreback) [inline pandas/core/frame.py]: should read Examples

**Comment by jorisvandenbossche** [comment]
Personally, I am not convinced that these examples are needed in the DataFrame docstring

- "Leaving dtype=None in constructor will infer a wider type than necessary" -> these are just the default dtypes, so I would phrase it in any case differently
- "A list/dict/Series/array-like is not allowed." -> here an example is certainly not needed IMO. We should just clarify in the explanation of the parameter above that "only a single dtype is allowed".

**Comment by jorisvandenbossche** [comment]
@patniharshit Can you update?

**Comment by patniharshit** [comment]
yeah sure

File: pandas/core/frame.py [hunk 2]
```diff
@@ -231,6 +231,21 @@ class DataFrame(NDFrame):
     >>> df3 = DataFrame(np.random.randn(10, 5),
     ...                 columns=['a', 'b', 'c', 'd', 'e'])
 
+    Leaving dtype=None in constructor will infer a wider type than necessary
```
Comment (jreback) [inline pandas/core/frame.py]: can you show an a typical example of passing a dtype, with a single type, then multiple types, then this example.

File: pandas/core/frame.py [hunk 3]
```diff
@@ -231,6 +231,21 @@ class DataFrame(NDFrame):
...
+    >>> df_cols = {'year':np.int32, 'month':np.int8}
+    >>> df = pd.DataFrame(columns=df_cols.keys(), dtype=None, index=range(10), data=-1)
+    >>> df.dtypes
+    month	int64
+    year	int64
+
+    A list/dict/Series/array-like is not allowed. Also behaves differently to read_csv(dtype)
```
Comment (jreback) [inline pandas/core/frame.py]: not sure what you are trying to show here, this is just confusing (after line 241)

**Comment by jreback** [comment]
can you rebase / update

**Comment by patniharshit** [comment]
I am extremely sorry to say that I won't be able to complete this now. Either someone take this up or feel free to close/delete this.

**Comment by jreback** [comment]
thanks @patniharshit np problem.

File: pandas/core/frame.py (hunk 2 above)
Comment (VincentLa) [inline pandas/core/frame.py]: What do you mean by an example passing multiple types? Is that possible to pass in multiple types?

**Pull request closed without merging**